Content.Server/Radar/RadarConsoleSystem.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Collections.Generic;
+using Content.Server.Solar.Components;
+using Content.Server.UserInterface;
+using Content.Shared.Radar;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Log;
+using Robust.Shared.Map;
+using Robust.Shared.Maths;
+
+namespace Content.Server.Radar;
+
+public class RadarConsoleSystem : EntitySystem
+{
+    [Dependency] private readonly IMapManager _mapManager = default!;
+
+    private static float Frequency = 1.5f;
+
+    private float _accumulator;
+
+    public override void Update(float frameTime)
+    {
+        _accumulator += frameTime;
+
+        if (_accumulator < Frequency)
+            return;
+
+        _accumulator -= Frequency;
+
+        foreach (var component in EntityManager.EntityQuery<RadarConsoleComponent>())
+        {
+            var s = component.Owner.GetUIOrNull(RadarConsoleUiKey.Key);
+            var radarTransform = Transform(component.Owner);
+            var radarPos = radarTransform.WorldPosition;
+            var radarRot = radarTransform.WorldRotation;
+
+            if (s is null)
+                continue;
+
+            var map = Transform(component.Owner).MapID;
+
+            var objects = new List<RadarObjectData>();
+
+            foreach (var grid in _mapManager.GetAllMapGrids(map))
+            {
+                var phy = Comp<PhysicsComponent>(grid.GridEntityId);
+                var transform = Transform(grid.GridEntityId);
+                if (phy.Mass < 50)
+                    continue;
+                var rad = Math.Log10(phy.Mass);
+                var pos = radarTransform.InvWorldMatrix.Transform(transform.WorldPosition);
```

Comment:
Cache InvWorldMatrix jesus.

Reply:
ALSO

It should use the centre of mass (LocalCenter on physicscomponent) because if someone maps say boxstation from the bottom-left then the origin will be significantly off.